### ISSUE TITLE
关于count_neg = tf.reduce_sum(1.0 - label)的问题

### ISSUE THREAD

**Issue by EricHuiK** [fengjian0106/hed-tutorial-for-document-scanning#79]
您好：
 count_neg = tf.reduce_sum(1.0 - label) # 样本中0的数量
        # count_neg = tf.reduce_sum(255.0 - label)  # 样本中0的数量
        count_pos = tf.reduce_sum(label) # 样本中1的数量(远小于count_neg)
        # print('debug, ==========================, count_pos is: '.format(count_pos),np.unique(count_pos))
        # print('debug, ==========================, count_neg is: '.format(count_neg),np.unique(count_neg))
        beta = count_neg / (count_neg + count_pos)  ## e.g.  60000 / (60000 + 800) = 0.9868

        pos_weight = beta / (1.0 - beta)  ## 0.986
这部分中看你的训练数据中label是二值化后的图片255表示边缘部分，label只有两个值0和255为什么1-label就是样本中0的数量呢？

**Comment by fengjian0106** [comment]
label: of the same shape. the ground truth in {0,1}.

label 里面的值是1和0

**Comment by EricHuiK** [comment]
我看到你的实例图是255和0，我发现我用同样的方式训练效果也不(我的label是0和255)，我与label是1的权重系数计算了对比了下，影响也不大




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "fengjian"<notifications@github.com&gt;; 
发送时间: 2020年9月23日(星期三) 中午11:25
收件人: "fengjian0106/hed-tutorial-for-document-scanning"<hed-tutorial-for-document-scanning@noreply.github.com&gt;; 
抄送: "流年啊你能耐我何"<875824970@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [fengjian0106/hed-tutorial-for-document-scanning] 关于count_neg = tf.reduce_sum(1.0 - label)的问题 (#79)





 
label: of the same shape. the ground truth in {0,1}.
 
label 里面的值是1和0

**Comment by fengjian0106** [comment]
def read_fix_size_image_format(dataset_root_dir_string, filename_queue)
这个函数里，有归一化处理

不管是不是 VGG，annotation 都需要归一化
annotation_float = annotation_float / 255.0

**Comment by EricHuiK** [comment]
对




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "fengjian"<notifications@github.com&gt;; 
发送时间: 2020年9月23日(星期三) 下午4:28
收件人: "fengjian0106/hed-tutorial-for-document-scanning"<hed-tutorial-for-document-scanning@noreply.github.com&gt;; 
抄送: "流年啊你能耐我何"<875824970@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [fengjian0106/hed-tutorial-for-document-scanning] 关于count_neg = tf.reduce_sum(1.0 - label)的问题 (#79)





 
def read_fix_size_image_format(dataset_root_dir_string, filename_queue)
 这个函数里，有归一化处理
 
不管是不是 VGG，annotation 都需要归一化
 
annotation_float = annotation_float / 255.0
